Add bookmarked_at to link

mAAdhaTTah opened the pull request:
I think this is pretty straightforward, but I don't know if you have any migration process set up.

mAAdhaTTah committed:
Add bookmarked_at to link

mAAdhaTTah committed:
Insert with bookmarked_at

This seems like it shouldn't be necessary, but the schema is only used
in the tests, not the production database.

mAAdhaTTah commented on db/schema.sql:
```diff
@@ -16,4 +16,5 @@ DEFINE FIELD url ON TABLE link TYPE string;
 DEFINE FIELD title ON TABLE link TYPE string;
 DEFINE FIELD note ON TABLE link TYPE string;
 DEFINE FIELD user ON TABLE link TYPE record (user);
+DEFINE FIELD bookmarked_at ON TABLE link TYPE datetime DEFAULT time::now();
```
I actually ran into an issue implementing this initially where it passed the tests but failed when actually using it because this schema only applies to tests, so the default I'm setting here wasn't getting used so I had to explicitly pass it in `create_link`.

joelseq replied:
I think that might be because currently the dev setup requires manually adding the schema definition (SurrealDB supports schemaless as well). I don't yet know if there is a good easy to use library for migrations yet so I currently just manually run these define statements. Either way I think I prefer the explicit assignment while creating the link!

mAAdhaTTah replied:
I started playing around with this a bit as I was looking into this problem which seemed pretty easy to integrate: https://github.com/Odonno/surrealdb-migrations/ I did run into an issue that required using their latest preview version, plus the way you have defined `surrealdb` as a dependency (pulling from github directly) conflicts with their dependency declaration (which caused some type mismatches), but I'd be happy to PR an integration with that if that would be helpful.

joelseq replied:
Yeah that would be greatly appreciated if you have time! :pray:

joelseq reviewed (approved):
The code LGTM! I'm just going to test some stuff out locally tomorrow before merging it in

mAAdhaTTah commented:
I swear I had the tests passing, looking at this.

mAAdhaTTah committed:
Switch to use surrealdb::sql::Datetime

chrono::DateTime<Utc> gets serialized to a string, not the type expected
by Surrealdb with the schema applied.

mAAdhaTTah commented:
Tests failing is almost certainly a result of the mismatch between tests + local. My best guess is because my local didn't apply the schema before running it, the `DateTime<Utc>` was cast to string and accepted and returned without issue but with the schema applied, it's having issues. _Could_ be related to this: https://github.com/surrealdb/surrealdb/issues/2804

Creating the `surrealdb::sql::Datetime` using `::from` seems to have solved it for the tests. Need to do a bit of testing on my local env to verify everything is good here.
